src/vorta/borg/create.py

```diff
@@ -149,6 +149,7 @@ def prepare(cls, profile):
                 pattern_file.write('\n'.join(exclude_dirs))
                 pattern_file.flush()
                 cmd.extend(['--exclude-from', pattern_file.name])
+                ret['cleanup_files'].append(pattern_file.name)
```

Comment:
```suggestion
                ret['pattern_temp_file'] = pattern_file
```

What about removing the `delete=False` when creating the temporary file and leaving it to the garbage collector? This will allow reusing a preparation object while removing the need for other changes.

Reply:
The idea was to be able to be more generic in case there are other temp files in the future. Or the extract job, which I didn't notice when looking at the code

Regarding delete=False, the file would be destroyed as soon as it is closed. Which I suppose we don't want, as this file is going to be passed later on to the borg command, right ?

Reply:
> The idea was to be able to be more generic in case there are other temp files in the future. Or the extract job, which I didn't notice when looking at the code

That is a good call. In that case keep in mind that one might want to implement cleaning up directories in the future.

> Regarding delete=False, the file would be destroyed as soon as it is closed. Which I suppose we don't want, as this file is going to be passed later on to the borg command, right ?

I think it is only closed when the python object is garbage collected.

Reply:
> That is a good call. In that case keep in mind that one might want to implement cleaning up directories in the future.
Yes, well, I suppose I could make it more complicated by having two arrays for temp_dirs, or just delete any name in the list, without distinction, which seems sufficient to me

> I think it is only closed when the python object is garbage collected.
Yes, so I suppose anytime after the end of the method, when pattern_file is out of scope, so it's going to be deleted, and we don't really know when, right ?

Reply:
> Yes, well, I suppose I could make it more complicated by having two arrays for temp_dirs, or just delete any name in the list, without distinction, which seems sufficient to me

One array is sufficient since it can store any kind of path.


> Agreed, so I suppose anytime after the end of the method, when pattern_file is out of scope and the destructor is going to be called, so it's going to be deleted, and we don't really know when, right ?

Exactly. However, if we store the file object in the `ret` dictionary, it won't be destructed until the dictionary is.

Reply:
> Exactly. However, if we store the file object in the ret dictionary, it won't be destructed until the dictionary is.

It feels really error prone, and hard to debug, just hoping it'll get destroyed when we want it, don't you think ? Really feels like using a side effect of us returning the name, and hoping the caller won't destroy ret too soon

Reply:
I think python garbage collection works well enough. Though your approach is very valid since it allows all kinds of leftover files.